e2e/interchaintest/types/ica_msg.go

```diff
@@ -127,3 +128,79 @@ func newSendCustomIcaMessagesMsg(cdc codec.BinaryCodec, msgs []proto.Message, en
 
 	return string(jsonBytes)
 }
+
+// newSendCosmosMsgsAsIcaTxMsg creates a new SendCosmosMsgsAsIcaTxMsg.
+func newSendCosmosMsgsAsIcaTxMsg(cosmosMsgs []ContractCosmosMsg, memo *string, timeout *uint64) string {
+	type SendCosmosMsgsAsIcaTxMsg struct {
+		Messages       []ContractCosmosMsg `json:"messages"`
+		PacketMemo     *string             `json:"packet_memo,omitempty"`
+		TimeoutSeconds *uint64             `json:"timeout_seconds,omitempty"`
+	}
+
+	type SendCosmosMsgsAsIcaTxMsgWrapper struct {
+		SendCosmosMsgsAsIcaTxMsg SendCosmosMsgsAsIcaTxMsg `json:"send_cosmos_msgs_as_ica_tx"`
+	}
+
+	msg := SendCosmosMsgsAsIcaTxMsgWrapper{
+		SendCosmosMsgsAsIcaTxMsg: SendCosmosMsgsAsIcaTxMsg{
+			Messages:       cosmosMsgs,
+			PacketMemo:     memo,
+			TimeoutSeconds: timeout,
+		},
+	}
+
+	jsonBytes, err := json.Marshal(msg)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(jsonBytes)
+}
+
+// newSendCosmosMsgsAsIcaTxMsg creates a new SendCosmosMsgsAsIcaTxMsg.
+func newSendCosmosMsgsAsIcaTxMsgFromProto(msgs []proto.Message, memo *string, timeout *uint64) string {
```

Comment:
This is not used in any of the tests yet. Create an issue to test every type of CosmosMsg